feconf.py

```diff
@@ -51,9 +51,13 @@
 TESTS_DATA_DIR = os.path.join('core', 'tests', 'data')
 SAMPLE_EXPLORATIONS_DIR = os.path.join('data', 'explorations')
 SAMPLE_COLLECTIONS_DIR = os.path.join('data', 'collections')
-INTERACTIONS_DIR = os.path.join('extensions', 'interactions')
-GADGETS_DIR = os.path.join('extensions', 'gadgets')
-RTE_EXTENSIONS_DIR = os.path.join('extensions', 'rich_text_components')
+
+EXTENSIONS_DIR_PREFIX = 'build_staging' if (IS_MINIFIED or not DEV_MODE) else ''
```

Comment:
Ah, I see.

Would you mind elaborating a bit on "the backend uses HTML files from _prod_"? I am wondering to what extent that is actually necessary. If we could transition (perhaps gradually) towards _prod_ not being necessary at all during runtime, that might be an improvement.

In the meantime, one idea is to remove core/templates/prod/head and have that in /build_staging/core/templates/dev/head (or something like that) so that all the prod files are in one place. That way we can say: here's the special "prod" folder, here's the special "build" folder, and everything else is dev. It would make the build process cleaner. Then over time perhaps we can get rid of prod altogether as we try to make the backend less dependent on files which should really be static anyway.